Tasks/PublishSymbolsV2/PreJobExecutionPublishSymbols.ts

```diff
@@ -0,0 +1,42 @@
+const path = require('path');
+const clientToolUtils = require('clienttool-common/ClientToolUtilities');
+const tl = require('azure-pipelines-task-lib');
+
+const toolName = "symbol";
+
+async function PreJobExecutionPublishSymbols(){
+    tl.setResourcePath(path.join(__dirname, "task.json"));
+    const UseNetCoreClientTool = tl.getBoolInput("UseNetCoreClientTool", false);
+
+    try {
+        // feature flag true or none Windows, then download NetCore version of client tool
+        if (UseNetCoreClientTool || tl.osType() != "Windows_NT") {
+
+            // Getting NetCore client tool
+            tl.debug("Getting NetCore client tool");
+            let clientToolFilePath = "/fake/path";
+
+            // Downloading the correct version of Symbol to target location
+            const accessToken = clientToolUtils.getSystemAccessToken();
+            const serviceUri = tl.getEndpointUrl("SYSTEMVSSCONNECTION", false);
+            const blobUri = await clientToolUtils.getBlobstoreUriFromBaseServiceUri(serviceUri, accessToken);
+
+            tl.debug(tl.loc("Info_RetrievingClientToolUri", blobUri));
+
+            // Downloading the client tool
+            clientToolFilePath = await clientToolUtils.retryOnExceptionHelper(
+                () => clientToolUtils.getClientToolFromService(
+                    blobUri,
+                    accessToken,
+                    toolName), 3, 1000);
+
+            tl.setTaskVariable('CLIENTTOOL_FILE_PATH', clientToolFilePath);
+        }
```

Comment:
What happens when someone is running on a Non-Windows agent and they don't select the UseNetCore option in the UI?

Reply:
Nvm, just saw the ||. hmm, An ideal way would be for us to automatically select and disable that option in UI, if it's a Non-Win machine.

Reply:
Not sure if this was discussed in requirement. WE can talk about this in meeting.

Reply:
Put this in info - "Used only on windows OS"

Reply:
In task.json, i have added this - 

"helpMarkDown": "Indicates whether to use version of the symbol upload tool that supports DWARF and ELF files. This option only matters on Windows agents. On non-Windows agents, the version of the symbol upload tool that supports DWARF and ELF files will always be used.",